docs/changes/coordinates/17046.api.rst

```diff
@@ -0,0 +1,2 @@
+The deprecated ``coordinates.get_moon()`` function has been removed. Use
+``coordinates.get_body("moon")`` instead.
```

Comment:
The change log entry will be in the `astropy.coordinates` section, so mentioning `coordinates` by name in the entry text is redundant, but we do have many entries that mention the sub-package by name already and having one more entry like that isn't too big of a deal.
```suggestion
The deprecated ``get_moon()`` function has been removed. Use
``get_body("moon")`` instead.
```